lanl/Draco
# Cleanup to allow Draco to build cleanly with new tools.

**Motivation**:
+ New Lmod modules will provide gcc/6.3.0 and openmpi/2.1.0 as defaults. We also have a new version of valgrind. This update eliminates some warnings that show up with the new tools.
+ Other unrelated cleanup is also included in this change set.

**Implementation**:
+ Significant change to `valgrind_suppress.txt` to address false positives reported by newer versions of gcc, openmpi, python, etc.
+ Register Draco's `valgrind_suppress.txt` file with cmake.    This allows developers to easily obtain valgrind test results from their build directories.
    
      *Example:* `ctest -D ExperimentalMemCheck [-j 8 -R c4]`

+ Update author list shown on Draco's github page.
+ Retire some build system code designed for older versions of OpenMPI.  The minimum required version of OpenMPI is now 1.8 (we use MPI 3.0 features).
+ Add some notes about clang sanitizer options.  We will depend more on these going forward since we will no longer run the `bounds_checking` nightly tests.
+ Mark the Compton unit test with `nomemcheck`.  This test takes too long when running under valgrind.
+ Minor formatting and documentation changes.
+ Refs #210 (GitHub).
* [Pre-Merge Code Review](https://github.com/lanl/Draco/wiki/Style-Guide)
  * [x] Travis CI checks pass
  * [x] Code coverage does not decrease
  * [x] [Valgrind test passes](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] [Toss2 checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] [Toss3 checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] [Trinitite checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [ ] Code reviewed/approved, sufficient DbC checks, testing, documentation
